### PR TITLE
Add Documentation Deployment Action

### PR DESCRIPTION
**Pull request by Javagedes**
Adds a github action that executes on a release that deploys the documentation to the pages site.

Example Execution on fork:

Action: https://github.com/Javagedes/edk2-pytool-extensions/actions/workflows/doc-release.yml

Site: https://javagedes.github.io/edk2-pytool-extensions/

### CODE REVIEW

File: docs/user/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+black==22.12.0
```
Comment (makubacki) [inline docs/user/requirements.txt:1]: Is this actually needed for the site publishing changes?

File: .github/workflows/doc-release.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      
+      - name: Build Documentation
+        run: |
+          mkdocs build --strict
+      
+      - name: Upload Github Pages Site Artifact
+        uses: actions/upload-pages-artifact@v1
+        with:
+          name: 'github-pages'
+          path: "site/"
```
Comment (makubacki) [inline .github/workflows/doc-release.yml:44]: Is there a pattern to the different quote styles being used? You don't need to change it, just curious why two simple strings close together have single and double quotes.

File: docs/user/requirements.txt (hunk 1 above)
Comment (makubacki) [reply docs/user/requirements.txt:1]: I noticed it is in #362 as well. I think it makes more sense to get in there.

**Review by makubacki — APPROVED** [review]

File: docs/user/requirements.txt (hunk 1 above)
Comment (Javagedes) [reply docs/user/requirements.txt:1]: Sounds good I'll merge that Pr first and remove it from here.

Black is used for some additional checks within mkdocs when building.

Comment (Javagedes) [reply docs/user/requirements.txt:1]: Removed

File: .github/workflows/doc-release.yml (hunk 2 above)
Comment (Javagedes) [reply .github/workflows/doc-release.yml:44]: Updated. They should match.

**Pull request merged**